dialogue-core/src/main/java/com/palantir/dialogue/core/AimdConcurrencyLimiter.java

```diff
@@ -35,23 +39,35 @@
     private final AtomicInteger limit = new AtomicInteger(INITIAL_LIMIT);
     private final AtomicInteger inFlight = new AtomicInteger();
 
-    Optional<Listener> acquire() {
+    /**
+     * Returns a new request token if the number of {@link #getInflight in-flight} tokens is smaller than the
+     * current {@link #getLimit upper limit} of allowed concurrent tokens. The caller is responsible for
+     * eventually releasing the token by calling exactly one of the {@link Token#ignore}, {@link Token#dropped},
+     * or {@link Token#success} methods.
+     *
+     * If the token
+     * is used in the context of a {@link Response Future&lt;Response&gt;} object, then passing the {@link Token} as a
+     * {@link FutureCallback callback} to the future will invoke either {@link Token#onSuccess} or
+     * {@link Token#onFailure} which delegate to
+     * ignore/dropped/success depending on the success or failure state of the response.
+     * */
+    Optional<Token> acquire() {
```

Comment:
could we use Permit instead of Token? Just because BearerToken is such a different concept here

Reply:
renamed to Permit